core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -365,6 +369,40 @@ public synchronized void losePrimacy() {
     LOG.info("Raft server successfully restarted");
   }
 
+  @Override
+  public synchronized Map<String, Long> getCurrentSequences() {
+    long currentGlobalState = mStateMachine.getLastAppliedSequenceNumber();
+    Map<String, Long> sequenceMap = new HashMap<>();
+    for (String master : mJournals.keySet()) {
+      // Return the same global sequence for each master.
+      sequenceMap.put(master, currentGlobalState);
+    }
+    return sequenceMap;
+  }
+
+  @Override
+  public synchronized void suspend() throws IOException {
+    // Suspend copycat snapshots.
+    mSnapshotAllowed.set(false);
+    mStateMachine.suspend();
+  }
+
+  @Override
+  public synchronized void resume() throws IOException {
+    mStateMachine.resume();
+    // Resume copycat snapshots.
+    mSnapshotAllowed.set(true);
```

Comment:
Is it safe to allow snapshots?

Reply:
The whole suspend/resume cycle is reserved for secondary role. So this is just restoring the original value after the backup.